Tests/Integration/Extend/Controller/ThankYouControllerTest.php

```diff
@@ -0,0 +1,60 @@
+<?php
+/**
+ * Shop System Plugins:
+ * - Terms of Use can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/_TERMS_OF_USE
+ * - License can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/LICENSE
+ */
+
+use OxidEsales\Eshop\Core\Registry;
+
+use PHPUnit\Framework\MockObject\MockObject;
+
+use Wirecard\Oxid\Extend\Controller\ThankYouController;
+
+class ThankYouControllerTest extends \Wirecard\Test\WdUnitTestCase
+{
+    /**
+     * @var ThankYouController|MockObject
+     */
+    private $_controller;
+
+    protected function setUp()
+    {
+        $this->_controller = oxNew(ThankYouController::class);
+        parent::setUp();
+    }
+
+    protected function dbData()
+    {
+        return [
+            [
+                'table' => 'oxorder',
+                'columns' => ['oxid', 'oxordernr'],
+                'rows' => [
+                    ['oxid1', '1'],
+                ],
+            ],
+        ];
+    }
+
+    public function testInit()
+    {
+        $oBasketStub = $this->getMockBuilder(\Wirecard\Oxid\Extend\Model\Basket::class)
+            ->disableOriginalConstructor()
+            ->setMethods(['getOrderId'])
+            ->getMock();
+
+        $oBasketStub->method('getOrderId')
+            ->willReturn('oxid1');
+
+        Registry::getSession()->setBasket($oBasketStub);
+
+        $this->_controller->init();
+
+        $this->assertNull(\OxidEsales\Eshop\Core\Registry::getSession()->getVariable('wdtoken'));
```

Comment:
The first assert is to check that we remove the token from the session because it has to be `consumed` in the init function.

we cannot split the test, there should be one test per method and there is a strict naming convention

see: https://docs.oxid-esales.com/developer/en/6.0/modules/certification/software_tests.html#test-classes

I understand the assertion "issue" but if you look at the oxid shop tests, they do the same. I can see no way to circumvent this.

Reply:
ok